plugins/package-managers/cocoapods/src/main/kotlin/PodDependencyHandler.kt

```diff
@@ -111,6 +115,43 @@ internal class PodDependencyHandler : DependencyHandler<Lockfile.Pod> {
         }
     }
 
+    private fun File.parsePodspecFile(): Podspec? {
+        val content = readText()
+
+        return if ("Pod::Spec.new" in content) {
+            // The podspec is in Ruby format.
+            // Because it may depend on React Native functions, an extra require my have to be injected.
+            val rubyContent = parentFile.searchUpwardsForSubdirectory("node_modules")?.let { nodeModulesParentDir ->
+                val reactNativePath =
+                    nodeModulesParentDir.resolve("node_modules/react-native/scripts/react_native_pods.rb")
+                if (reactNativePath.isFile) {
+                    "require '$reactNativePath'\n$content"
+                } else {
+                    null
+                }
+            } ?: content
+
+            val tempRubyPath = createTempFile("ruby_podspec", ".podspec").also { it.writeText(rubyContent) }
+
+            val podspecProcess = CocoaPodsCommand.run(
```

Comment:
Please add a code comment what this does, even if mentioned in the commit message.